### DISCUSSION TITLE
Route error

### DISCUSSION THREAD

**Discussion by niubiworker** [gdsfactory/gdsfactory#3213] [Q&A]
I have defined a function to implement a boolean operation on routing paths, but when I use these two sets of ports, the routing has obvious errors. I would like to know why? gdsfactory=7.27, python=3.11.
![屏幕截图 2024-09-20 114700](https://github.com/user-attachments/assets/0f0a7da7-776d-4a6a-a12b-d8b32b902f40)
![image](https://github.com/user-attachments/assets/83597a1e-387a-4d42-a070-0f14d5529b46)
```
@gf.cell
def portconnect(ports1_group,ports2_group,width:float=12,str_width:float=1):
    c=gf.Component()
    routes = gf.routing.get_bundle(
    ports1_group,
    ports2_group,
    start_straight_length=100,
    radius=100,width=width,separation=0
)
    routes1 = gf.routing.get_bundle(
    ports1_group,
    ports2_group,
    start_straight_length=100,
    radius=100,width=str_width,separation=0
)
    for route, route1 in zip(routes, routes1):
        bool_result = gf.geometry.boolean(
            A=route.references,
            B=route1.references,
            operation="not",
            precision=1e-9,
            layer=(3, 0)
        )
        
        c.add(bool_result)
    return c
```
thank you very much！

**Comment by sebastian-goeldi** [comment]
I am pretty sure you need to sort ports, either manually, or tell the route_bundle function to do so (there's a flag afaik). Looking at your port locations, you are connecting them in reversed order.

I am not a gdsfactory7 expert though ;)
